src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleFormatter.cs

```diff
@@ -79,10 +77,16 @@ public override void Write<TState>(in LogEntry<TState> logEntry, IExternalScopeP
 
                     if (logEntry.State is IReadOnlyCollection<KeyValuePair<string, object>> stateDictionary)
```

Comment:
>IReadOnlyCollection [](start = 42, length = 19)

does it make sense to use IEnumerable rather than IReadOnlyCollection? this will give more coverage to other types.

Reply:
I don't think it is supposed to work if it is not IReadOnlyCollection. Refer to https://github.com/dotnet/runtime/issues/34742#issuecomment-660708530

Reply:
So why we are testing the type? doesn't mean you can get other types?

Reply:
From my understanding the contract expects IReadOnlyCollection<KeyValuePair<string, object>> and that is why I check for it. (Referring to @davidfowl 's comment in https://github.com/dotnet/runtime/issues/34742#issuecomment-660708530)